### ISSUE TITLE
Warning on Wayland when I run the `ydotoold` daemon: "WARNING: running xinput against an Xwayland server." "unable to find device pointer:ydotoold virtual device"

### ISSUE THREAD

**Issue by ElectricRCAircraftGuy** [ReimuNotMoe/ydotool#205]
I just built `ydotool` from commit 57ba7d0af525e82da2de0e275d169477f293b197 on Ubuntu 22.04.2. `ydotoold --version` shows `v1.0.4-0-g57ba7d0`. 

When running the Wayland window manager (see how to select X11 vs Wayland [in my answer here](https://askubuntu.com/a/1470563/327339)), I get the following warning when I start the daemon:

```
WARNING: running xinput against an Xwayland server. See the xinput man page for details.
unable to find device pointer:ydotoold virtual device
```

What does this mean and what implications does it have? What about `unable to find device pointer:ydotoold virtual device`? Does that mean I am unable to fully control the mouse with `ydotool`?

Here is the full command and output:
```bash
$ sudo ydotoold --socket-path="$HOME/.ydotool_socket" --socket-own="$(id -u):$(id -g)"
[sudo] password for gabriel: 
Socket path: /home/gabriel/.ydotool_socket
Removing old stale socket
Socket permission: 0600
Socket ownership: UID=1000, GID=1000
READY
WARNING: running xinput against an Xwayland server. See the xinput man page for details.
unable to find device pointer:ydotoold virtual device
```

**Comment by hrqmonteiro** [comment]
No one to look at this?

**Comment by Paiusco** [comment]
Can you retry that with latest main and see if this still happens @ElectricRCAircraftGuy? Currently don't have a ubuntu machine, but we could try to tackle it

**Comment by ElectricRCAircraftGuy** [comment]
(I'm still following this but unable to try it again at this time).

**Comment by DmitryDGD** [comment]
@Paiusco  Manjaro KDE plasma 5.27.11, Wayland. Same thing! Win+D works.
![2024-05-09_18-22](https://github.com/ReimuNotMoe/ydotool/assets/60581708/a2a3bb97-4713-435d-92aa-eb439ae39dab)

**Comment by Paiusco** [comment]
@DmitryDGD 
oh, but the thing is that you defined the socket for the daemon, you have to do that on the cli as well. Otherwise the client will go for the default path, which is `/run/user/...`

So the way you started the daemon is fine, but as the **last line from `ydotool --help`** says:

_Use environment variable YDOTOOL_SOCKET to specify daemon socket_

so you should try to run as:
`YDOTOOL_SOCKET="$HOME/.ydotool.socket" ydotool type`

**Comment by Paiusco** [comment]
Ok, I tracked this issue introduced by this request: #164.
I need some help from people that has `xinput` both on X and XWayland real installation as I don't have time to do that now. (I use X and don't have `xinput` installed. I'll show my XWayland result from a VM)

So maybe it's also desirable to disable the mouse acceleration on XWayland (maybe @ReimuNotMoe can give some _inputs_ here, pun intended), even tho that may only work for that session... 

If someone can:

1. start `ydotoold`
2. run `xinput list` and post them here

So I can try to get any pattern there... What I can see on my XWayland VM is:
![image](https://github.com/ReimuNotMoe/ydotool/assets/2374887/2e0078ae-ce2b-4048-983e-923c0d9d5e1a)
so the property that `ydotoold` tries to change doesn't exist indeed, and the message makes sense.
----

@ReimuNotMoe questions:

1. Does it make sense to try to disable the mouse acceleration on XWayland? `xinput` states that: 
![image](https://github.com/ReimuNotMoe/ydotool/assets/2374887/1faeed23-46d0-4b6d-bdbd-2a0092295e12)

I think we should remove this from XWayland cases... They mention we should use the compositor's tools for it, that may give us a headache, as every compositor will probably have their own way, no?

2. Is the naming always the same for the property on X? I mean [this line](https://github.com/ReimuNotMoe/ydotool/blob/ac762710856c8b4bf6c217efd93f7fe61c923e9b/Daemon/ydotoold.c#L365). Is it **always** `pointer:ydotoold virtual device`?
 
I'll implement a fix once we define what to do

**Comment by idsamcom** [comment]
It seems that the warning message can be ignored. But I have to follow @jonas73x's comment [here](https://github.com/ReimuNotMoe/ydotool/issues/245#issuecomment-2291798216) to make it work on Wayland. It is weird because everything works on X11 without any changes.

**Comment by Paiusco** [comment]
@idsamcom thank you for your tests, I've opened #255 to fix it. Tested under wayland and indeed doesn't hurt, so the easiest approach seems to make it silent for now. Hopefully will be merged soon

**Comment by KavyanshKhaitan2** [comment]
@Paiusco this still keeps happening on Ubuntu 24.04. I tested on a fresh VM that I used to test https://github.com/asweigart/pyautogui/pull/936. The tests kept failing because ydotool didnt move the mouse cursor.

Please look into it ASAP!

(Edit: I dont really have much low-level knowledge, otherwise I would be free to contribute to ydotool aswell!)
```
user@ubuntuvm:~$ xinput list
WARNING: running xinput against an Xwayland server. See the xinput man page for details.
⎡ Virtual core pointer                    	id=2	[master pointer  (3)]
⎜   ↳ Virtual core XTEST pointer              	id=4	[slave  pointer  (2)]
⎜   ↳ xwayland-pointer:14                     	id=6	[slave  pointer  (2)]
⎜   ↳ xwayland-relative-pointer:14            	id=7	[slave  pointer  (2)]
⎜   ↳ xwayland-pointer-gestures:14            	id=8	[slave  pointer  (2)]
⎣ Virtual core keyboard                   	id=3	[master keyboard (2)]
    ↳ Virtual core XTEST keyboard             	id=5	[slave  keyboard (3)]
    ↳ xwayland-keyboard:14                    	id=9	[slave  keyboard (3)]
user@ubuntuvm:~$ 

```

**Comment by Paiusco** [comment]
> [@Paiusco](https://github.com/Paiusco) this still keeps happening on Ubuntu 24.04. I tested on a fresh VM that I used to test [asweigart/pyautogui#936](https://github.com/asweigart/pyautogui/pull/936). The tests kept failing because ydotool didnt move the mouse cursor.
> 
> Please look into it ASAP!

Have couple of PRs opened here for months, no response... Don't know if it'd fully fix your case though. The maintainers are working on rewriting the whole thing in JavaScript anyway. (See their readme)
So I'm not spending much time here until they are back.

**Comment by KavyanshKhaitan2** [comment]
> The maintainers are working on rewriting the whole thing in JavaScript anyway. (See their readme) So I'm not spending much time here until they are back.

Yeah, I did see the readme before commenting here (maybe theres a fix for it there?).

I dont like how its happening behind closed doors. Couldnt find any other repo for the new rewrite. Would be better if @ReimuNotMoe made it public (👀).

Also didnt find any mentions of the "ultra-light JS runtime" (Resonance) outside of this project's README, which states it should already be released (in Q2 2024).

I honestly dont believe the rewrite is coming any-time soon, if ever.

**Comment by gcqmkm02** [comment]
wayland mutter completely broken
